tls/s2n_handshake_io.c

```diff
@@ -712,13 +792,19 @@ int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status * blocked)
             }
         } else {
             *blocked = S2N_BLOCKED_ON_READ;
-            if (handshake_read_io(conn) < 0) {
+            int r = handshake_read_io(conn);
+            if (r < 0) {
```

Comment:
Here and line 758 are same logic we can put in a common function like s2n_try_delete_session_cache